implementations/rust/ockam/ockam_entity/src/entity.rs

```diff
@@ -0,0 +1,363 @@
+use crate::{
+    Contact, ContactsDb, KeyAttributes, ProfileAuth, ProfileChangeEvent, ProfileChanges,
+    ProfileContacts, ProfileEventAttributes, ProfileIdentifier, ProfileIdentity, ProfileImpl,
+    ProfileSecrets, ProfileTrait, ProfileVault, ProfileVaultAccess,
+};
+
+use crate::EntityError::ProfileNotFound;
+use ockam_core::Result;
+use ockam_vault::ockam_vault_core::{PublicKey, Secret};
+
+/// An Entity represents an identity in various authentication contexts.
+#[derive(Clone)]
+pub struct Entity<V: ProfileVault> {
+    default_profile_identifier: ProfileIdentifier,
```

Comment:
We're storing a copy of the profile identifier, instead of a ref to one in the set. A ref would require introducing a lifetime which we don't want to do. This also allows the underlying Profile to be lazily loaded and otherwise decoupled from the profile impl storage. That being said, A profile with this identifier must be in the `profiles` set. Since a default profile is required on `Entity` construction, we can guarantee this if we forbid removal of the default profile, when we add profile removal.